django_airavata/apps/api/views.py

```diff
@@ -709,6 +713,92 @@ def get(self, request, format=None):
             data_product, context={'request': request})
         return Response(serializer.data)
 
+experiment_data_storage = FileSystemStorage(
+    location=settings.GATEWAY_DATA_STORE_DIR)
+
+@login_required
+def get_user_files(request):
+    try:
+
+        dirs=[] # a list with file_name and file_dpu for each file
+
+        for o in User_Files.objects.values_list('file_name','file_dpu'):
+            file_details={}
+            file_details['file_name']=o[0];
+            file_details['file_dpu']=o[1];
+            dirs.append(file_details);
+
+        dirs_result=json.dumps(dirs)
```

Comment:
You don't need to dump this to a string. You can simply return `return JsonResponse({'uploaded': True,'user-files':dirs})` and then you don't need to do JSON.parse in UserFileInputEditor.vue.